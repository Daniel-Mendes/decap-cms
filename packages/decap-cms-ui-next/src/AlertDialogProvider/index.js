export { AlertDialogProvider, useAlertDialogContext } from './AlertDialogProvider';
