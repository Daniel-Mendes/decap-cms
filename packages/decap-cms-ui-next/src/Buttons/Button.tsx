import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/react';
import color from 'color';

import Icon from '../Icon';
import ButtonGroup from './ButtonGroup';

import type { IconName } from '../Icon';

export const StyledButton = styled.button<ButtonProps>`
  ${({ theme, size, type, variant, transparent, disabled }) => css`
    font-family: ${theme.fontFamily};
    font-size: ${size === 'sm' ? 12 : size === 'lg' ? 14 : 14}px;
    padding: ${size === 'sm' ? `0 0.5rem` : size === 'lg' ? `0 1rem` : `0 0.75rem`};
    line-height: ${size === 'sm' ? `1.5rem` : size === 'lg' ? `2.5rem` : `2rem`};
    font-weight: 600;
    text-align: center;
    border-radius: ${size === 'sm' ? 4 : 6}px;
    border: 0;
    outline: 0;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};
    white-space: nowrap;
    transition: 200ms;
    background-color: ${transparent
      ? 'transparent'
      : variant === 'solid'
      ? type === 'success'
        ? theme.color.success['900']
        : type === 'danger'
        ? theme.color.danger['900']
        : type === 'neutral'
        ? theme.color.neutral[theme.darkMode ? '800' : '800']
        : theme.color.primary[900]
      : type === 'success'
      ? color(theme.color.success['900']).alpha(0.1).string()
      : type === 'danger'
      ? color(theme.color.danger['900']).alpha(0.1).string()
      : type === 'neutral'
      ? color(theme.color.neutral['700']).alpha(0.1).string()
      : color(theme.color.primary['900']).alpha(0.1).string()};
    color: ${variant === 'solid'
      ? '#FFFFFF'
      : type === 'success'
      ? theme.color.success[theme.darkMode ? '200' : '700']
      : type === 'danger'
      ? theme.color.danger[theme.darkMode ? '200' : '700']
      : type === 'neutral'
      ? theme.color.neutral[theme.darkMode ? '300' : '1000']
      : theme.color.primary[theme.darkMode ? '200' : '900']};
    ${disabled
      ? 'opacity: 0.5;'
      : `
          &:hover {
            background-color: ${
              transparent
                ? 'rgba(0,0,0,0.15)'
                : variant === 'solid'
                ? type === 'success'
                  ? theme.color.success['800']
                  : type === 'danger'
                  ? theme.color.danger['800']
                  : type === 'neutral'
                  ? theme.color.neutral['700']
                  : theme.color.primary['800']
                : type === 'success'
                ? color(theme.color.success['900'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
                : type === 'danger'
                ? color(theme.color.danger['900'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
                : type === 'neutral'
                ? color(theme.color.neutral['700'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
                : color(theme.color.primary['900'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
            };
          }
          &:focus,
          &:focus:hover {
            background-color: ${
              variant === 'solid'
                ? type === 'success'
                  ? theme.color.success['800']
                  : type === 'danger'
                  ? theme.color.danger['800']
                  : type === 'neutral'
                  ? theme.color.neutral['700']
                  : theme.color.primary['800']
                : type === 'success'
                ? color(theme.color.success['900'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
                : type === 'danger'
                ? color(theme.color.danger['900'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
                : type === 'neutral'
                ? color(theme.color.neutral['700'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
                : color(theme.color.primary['900'])
                    .alpha(theme.darkMode ? 0.35 : 0.2)
                    .string()
            };
          }
          &:active,
          &:active:focus {
            background-color: ${
              variant === 'solid'
                ? type === 'success'
                  ? theme.color.success['1000']
                  : type === 'danger'
                  ? theme.color.danger['1000']
                  : type === 'neutral'
                  ? theme.color.neutral['900']
                  : theme.color.primary['1000']
                : type === 'success'
                ? color(theme.color.success['500'])
                    .alpha(theme.darkMode ? 0.2 : 0.35)
                    .string()
                : type === 'danger'
                ? color(theme.color.danger['900'])
                    .alpha(theme.darkMode ? 0.2 : 0.35)
                    .string()
                : type === 'neutral'
                ? color(theme.color.neutral['700'])
                    .alpha(theme.darkMode ? 0.2 : 0.35)
                    .string()
                : color(theme.color.primary['900'])
                    .alpha(theme.darkMode ? 0.2 : 0.35)
                    .string()
            };
          }
          `}
    ${ButtonGroup} & {
      margin: 4px;
    }
  `}
  ${({ type, theme, iconOnly }) =>
    iconOnly
      ? `
      color: ${theme.color.neutral['700']};
      padding: 0 0.5rem;
      &,
      &:hover,
      &:focus,
      &:active,
      &:focus:hover,
      &:focus:active {
        background-color: transparent;
      }
      &:hover {
        color: ${type === 'danger' ? theme.color.danger['500'] : theme.color.primary['500']}
      }
  `
      : ''}
`;

const StyledMenuIcon = styled(Icon)`
  display: inline-block;
  vertical-align: middle;
  margin-left: 0.5rem;
`;

const StyledIcon = styled(Icon)`
  ${({ hasText, size }) =>
    hasText ? `margin-right: ${size === 'sm' ? 0.25 : size === 'lg' ? 0.75 : 0.5}rem;` : ''}
  ${({ hasText, size }) =>
    hasText ? `margin-left: -${size === 'sm' ? 0.125 : size === 'lg' ? 0.375 : 0.25}rem;` : ''}
  margin-top: -${({ size }) => (size === 'sm' ? 1.5 : size === 'lg' ? 3 : 2)}px;
  margin-bottom: -${({ size }) => (size === 'sm' ? 0.75 : size === 'lg' ? 1.5 : 1)}px;
  vertical-align: middle;
`;

type ButtonProps = {
  icon?: IconName;
  children?: React.ReactNode;
  className?: string;
  hasMenu?: boolean;
  size?: 'sm' | 'md' | 'lg';
  type?: 'primary' | 'success' | 'danger' | 'neutral';
  variant?: 'solid' | 'soft';
  disabled?: boolean;
  transparent?: boolean;
};

function Button({
  icon,
  children,
  className,
  hasMenu,
  size = 'md',
  type = 'primary',
  variant = 'solid',
  disabled,
  ...props
}: ButtonProps) {
  return (
    <StyledButton
      size={size}
      type={type}
      variant={variant}
      iconOnly={!children && icon}
      className={className}
      disabled={disabled}
      {...props}
    >
      {icon && <StyledIcon hasText={!!children} name={icon} size={size === 'lg' ? 'md' : size} />}
      {children}
      {hasMenu && <StyledMenuIcon size={size === 'lg' ? 'md' : size} name="chevron-down" />}
    </StyledButton>
  );
}

export default Button;
