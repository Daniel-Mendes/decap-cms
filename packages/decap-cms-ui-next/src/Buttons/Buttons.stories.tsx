import React from 'react';
import { userEvent, waitFor, within, expect, fn } from '@storybook/test';

import { Button, ButtonGroup, IconButton, AvatarButton, FileUploadButton } from '.';
import { iconComponents } from '../Icon';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Button> = {
  title: 'Components/Button',
  component: Button,
  args: {
    children: 'Button',
    onClick: fn(),
  },
};

export default meta;
type Story = StoryObj<typeof Button>;

// export const Default: Story = {
//   play: async ({ args, canvasElement, step }) => {
//     const canvas = within(canvasElement);

//     await step('Click button', async () => {
//       const button = canvas.getByRole('button');
//       await userEvent.click(button);
//     });

//     // assert
//     expect(args.children).toBe('Button');
//     await waitFor(() => expect(args.onClick).toHaveBeenCalled());
//   },
// };

export const Default: Story = {};

export const Neutral: Story = {
  args: {
    type: 'neutral',
  },
};

export const Success: Story = {
  args: {
    type: 'success',
  },
};

export const Danger: Story = {
  args: {
    type: 'danger',
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
  },
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Click button', async () => {
      const button = canvas.getByRole('button');
      await userEvent.click(button);

      expect(args.onClick).not.toHaveBeenCalled();
    });
  },
};

export const WithIcon: Story = {
  args: {
    icon: 'plus',
  },
};

export const WithMenu: Story = {
  args: {
    hasMenu: true,
  },
};

export const Transparent: Story = {
  args: {
    transparent: true,
  },
};

export const Solid: Story = {
  args: {
    variant: 'solid',
  },
};

export const Soft: Story = {
  args: {
    variant: 'soft',
  },
};

export const Small: Story = {
  args: {
    size: 'sm',
  },
};

export const Medium: Story = {
  args: {
    size: 'md',
  },
};

export const Large: Story = {
  args: {
    size: 'lg',
  },
};

export const ClickInteraction: Story = {
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Click button', async () => {
      const button = canvas.getByRole('button');
      await userEvent.click(button);

      expect(args.onClick).toHaveBeenCalled();
    });
  },
};

// Render all variants of the button in a single story to take only one Image snapshot
export const ImageSnapshot: Story = {
  render: ({ args }) => (
    <ButtonGroup>
      <Button {...args}>Primary Button</Button>
      <Button {...args} type="success">
        Success Button
      </Button>
      <Button {...args} type="danger">
        Danger Button
      </Button>
      <Button {...args} type="neutral">
        Neutral Button
      </Button>
    </ButtonGroup>
  ),
};

// export function _Button(args) {
//   return <Button {...args}>Button</Button>;
// }

// _Button.argTypes = {
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//     table: {
//       defaultValue: { summary: 'md' },
//     },
//   },
//   type: {
//     control: 'select',
//     options: ['primary', 'success', 'danger', 'neutral'],

//     table: {
//       defaultValue: { summary: 'primary' },
//     },
//   },
//   variant: {
//     control: 'select',
//     options: ['solid', 'soft'],
//     table: {
//       defaultValue: { summary: 'solid' },
//     },
//   },
//   disabled: {
//     control: 'boolean',
//     table: {
//       defaultValue: { summary: 'false' },
//     },
//   },
//   transparent: {
//     control: 'boolean',
//     table: {
//       defaultValue: { summary: 'false' },
//     },
//   },
//   icon: {
//     control: 'select',
//     options: {
//       default: null,
//       ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
//     },
//     table: {
//       defaultValue: { summary: 'null' },
//     },
//   },
// };

// _Button.args = {
//   type: 'primary',
//   size: 'md',
//   variant: 'solid',
//   disabled: false,
//   transparent: false,
//   icon: null,
//   onClick: fn(),
// };

// export function _ButtonGroup(args) {
//   return (
//     <ButtonGroup {...args}>
//       <Button>Button 1</Button>
//       <Button>Button 2</Button>
//       <Button primary type="success">
//         Primary Success Button
//       </Button>
//     </ButtonGroup>
//   );
// }

// _ButtonGroup.argTypes = {
//   direction: {
//     control: 'select',
//     options: ['horizontal', 'vertical'],
//     mapping: {
//       horizontal: null,
//     },
//   },
// };

// _ButtonGroup.args = {
//   direction: 'horizontal',
// };

// export function _IconButton(args) {
//   return <IconButton {...args} />;
// }

// _IconButton.argTypes = {
//   icon: {
//     control: 'select',
//     options: {
//       default: null,
//       ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
//     },
//   },
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//   },
//   active: {
//     control: 'boolean',
//   },
// };

// _IconButton.args = {
//   icon: null,
//   size: 'md',
//   active: false,
//   onClick: fn(),
// };

// export function _AvatarButton(args) {
//   return <AvatarButton {...args} />;
// }

// _AvatarButton.argTypes = {
//   src: {
//     control: 'text',
//   },
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//   },
//   active: {
//     control: 'boolean',
//   },
// };

// _AvatarButton.args = {
//   src: 'https://randomuser.me/api/portraits/men/94.jpg',
//   size: 'md',
//   active: false,
//   onClick: fn(),
// };

// export function _FileUploadButton(args) {
//   return <FileUploadButton {...args} />;
// }

// _FileUploadButton.argTypes = {
//   label: {
//     control: 'text',
//   },
//   accept: {
//     control: 'select',
//     options: ['audio/*', 'image/*', 'video/*', '*/*'],
//   },
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//   },
//   disabled: {
//     control: 'boolean',
//     table: {
//       defaultValue: { summary: 'false' },
//     },
//   },
// };

// _FileUploadButton.args = {
//   label: 'Upload File',
//   accept: 'image/*',
//   size: 'md',
//   disabled: false,
//   onChange: fn(),
// };
