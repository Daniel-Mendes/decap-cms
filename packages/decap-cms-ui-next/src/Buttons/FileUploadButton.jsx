import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

import { Button } from './';

const StyledButton = styled(Button)`
  margin: 0 !important;
`;

const HiddenInputFile = styled.input`
  height: 0.1px;
  width: 0.1px;
  margin: 0;
  padding: 0;
  opacity: 0;
  overflow: hidden;
  position: absolute;
  z-index: -1;
  outline: none;
`;

function FileUploadButton({
  label,
  icon = 'upload-cloud',
  size,
  disabled,
  accept,
  onChange,
  ...props
}) {
  const inputRef = useRef(null);

  return (
    <label>
      <StyledButton
        size={size}
        icon={icon}
        disabled={disabled}
        onClick={() => inputRef.current.click()}
        {...props}
      >
        {label}
      </StyledButton>

      <HiddenInputFile
        type="file"
        ref={inputRef}
        accept={accept}
        onChange={onChange}
        disabled={disabled}
      />
    </label>
  );
}

FileUploadButton.propTypes = {
  label: PropTypes.string.isRequired,
  icon: PropTypes.string,
  size: PropTypes.oneOf(['sm', 'md', 'lg']),
  disabled: PropTypes.bool,
  accept: PropTypes.oneOf(['audio/*', 'image/*', 'video/*', '*/*']),
  onChange: PropTypes.func.isRequired,
};

export default FileUploadButton;
