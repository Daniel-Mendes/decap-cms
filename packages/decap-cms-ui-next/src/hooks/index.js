export { default as useLocalStorageState } from './localStorageState';
