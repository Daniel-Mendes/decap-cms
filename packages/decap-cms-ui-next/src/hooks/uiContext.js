import { useContext } from 'react';

import { UIContext } from '../UIContext';

export function useUIContext() {
  return useContext(UIContext);
}
