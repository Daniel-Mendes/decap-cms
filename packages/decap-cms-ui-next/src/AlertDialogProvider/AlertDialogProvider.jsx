import React, { useContext, createContext, useState } from 'react';
import styled from '@emotion/styled';

import Dialog from '../Dialog';
import { Button, ButtonGroup } from '../Buttons';
import { TextField } from '../Fields';
import { result } from 'lodash';

const DialogContent = styled.div`
  display: flex;
  flex-direction: column;
  gap: 1rem;
`;

const AlertDialogContext = createContext();

export function useAlertDialogContext() {
  const context = useContext(AlertDialogContext);

  if (!context) {
    throw new Error('useAlertDialogContext must be used within a AlertDialogProvider');
  }

  return context;
}

export function AlertDialogProvider({ children }) {
  const [dialogConfig, setDialogConfig] = useState({
    open: false,
    type: '',
    title: '',
    message: '',
    defaultValue: '',
    promptInputValue: '',
    actionButton: '',
    cancelButton: '',
    onConfirm: null,
    onCancel: null,
  });

  function openDialog({ type, title, message, defaultValue, actionButton, cancelButton }) {
    return new Promise((resolve, reject) => {
      setDialogConfig({
        open: true,
        type,
        title,
        message,
        defaultValue,
        promptInputValue: defaultValue || '',
        actionButton,
        cancelButton,
        onConfirm: result => resolve(result),
        onCancel: () => resolve(false),
      });
    });
  }

  function closeDialog() {
    setDialogConfig({ ...dialogConfig, open: false });
  }

  function handleConfirm() {
    if (dialogConfig.onConfirm) {
      if (dialogConfig.type === 'prompt') {
        dialogConfig.onConfirm(dialogConfig.promptInputValue);
      } else {
        dialogConfig.onConfirm(true);
      }
    }
    closeDialog();
  }

  function handleCancel() {
    if (dialogConfig.onCancel) dialogConfig.onCancel();
    closeDialog();
  }

  function handlePromptInputChange(event) {
    setDialogConfig({ ...dialogConfig, promptInputValue: event.target.value });
  }

  return (
    <AlertDialogContext.Provider
      value={{
        open: dialogConfig.open,
        type: dialogConfig.type,
        title: dialogConfig.title,
        message: dialogConfig.message,
        defaultValue: dialogConfig.defaultValue,
        actionButton: dialogConfig.actionButton,
        cancelButton: dialogConfig.cancelButton,
        onConfirm: dialogConfig.onConfirm,
        onCancel: dialogConfig.onCancel,
        openDialog,
      }}
    >
      {children}

      <Dialog
        open={dialogConfig.open}
        title={dialogConfig.title}
        onClose={closeDialog}
        disableBackdropClick={true}
        actions={
          dialogConfig.type === 'alert' ? (
            <Button onClick={handleConfirm}>{dialogConfig.actionButton}</Button>
          ) : dialogConfig.type === 'confirm' ? (
            <ButtonGroup>
              <Button onClick={handleCancel} type="danger">
                {dialogConfig.cancelButton}
              </Button>
              <Button type="success" onClick={handleConfirm}>
                {dialogConfig.actionButton}
              </Button>
            </ButtonGroup>
          ) : dialogConfig.type === 'prompt' ? (
            <ButtonGroup>
              <Button onClick={handleCancel} type="danger">
                {dialogConfig.cancelButton}
              </Button>
              <Button type="success" onClick={handleConfirm}>
                {dialogConfig.actionButton}
              </Button>
            </ButtonGroup>
          ) : null
        }
      >
        <DialogContent>
          {dialogConfig.message}

          {dialogConfig.type === 'prompt' && (
            <TextField
              type="text"
              value={dialogConfig.promptInputValue}
              onChange={handlePromptInputChange}
              focus
            />
          )}
        </DialogContent>
      </Dialog>
    </AlertDialogContext.Provider>
  );
}
