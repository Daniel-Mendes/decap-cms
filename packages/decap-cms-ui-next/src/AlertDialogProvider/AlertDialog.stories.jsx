import React from 'react';
import { action } from '@storybook/addon-actions';

import { useAlert, useConfirm, usePrompt } from '../hooks/useAlertDialog';
import { Button } from '../Buttons';
import { AlertDialogProvider } from './AlertDialogProvider';

export default {
  title: 'hooks/useAlertDialog',
  decorators: [
    Story => (
      <AlertDialogProvider>
        <Story />
      </AlertDialogProvider>
    ),
  ],
};

export function _useAlert(args) {
  const alert = useAlert();

  return <Button onClick={() => alert(args)}>Alert Dialog</Button>;
}

_useAlert.args = {
  title: 'Alert Dialog Title',
  message: 'This is an alert dialog message',
  actionButton: 'OK',
};

export function _useConfirm(args) {
  const confirm = useConfirm();

  return <Button onClick={() => confirm(args)}>Confirm Dialog</Button>;
}

_useConfirm.args = {
  title: 'Confirm Dialog Title',
  message: 'This is a confirm dialog message',
  actionButton: 'OK',
  cancelButton: 'Cancel',
};

export function _usePrompt(args) {
  const prompt = usePrompt();

  return <Button onClick={() => prompt(args)}>Prompt Dialog</Button>;
}

_usePrompt.args = {
  title: 'Prompt Dialog Title',
  message: 'This is a prompt dialog message',
  defaultValue: 'Default value',
  actionButton: 'OK',
  cancelButton: 'Cancel',
};
