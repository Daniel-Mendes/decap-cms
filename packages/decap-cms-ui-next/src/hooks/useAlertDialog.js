import { useAlertDialogContext } from '../AlertDialogProvider';

export function useAlert() {
  const { openDialog } = useAlertDialogContext();

  return async ({ title, message, actionButton }) =>
    await openDialog({ type: 'alert', title, message, actionButton });
}

export function useConfirm() {
  const { openDialog } = useAlertDialogContext();

  return async ({ title, message, actionButton, cancelButton }) =>
    await openDialog({
      type: 'confirm',
      title,
      message,
      actionButton,
      cancelButton,
    });
}

export function usePrompt() {
  const { openDialog } = useAlertDialogContext();

  return async ({ title, message, defaultValue, actionButton, cancelButton }) =>
    await openDialog({
      type: 'prompt',
      defaultValue,
      title,
      message,
      actionButton,
      cancelButton,
    });
}
