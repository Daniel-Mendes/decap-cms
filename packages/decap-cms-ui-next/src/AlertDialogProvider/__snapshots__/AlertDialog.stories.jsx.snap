// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks/useAlertDialog _useAlert smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Alert Dialog</button>"`;

exports[`hooks/useAlertDialog _useConfirm smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Confirm Dialog</button>"`;

exports[`hooks/useAlertDialog _usePrompt smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Prompt Dialog</button>"`;
